cpp/CMakePresets.json

```diff
@@ -371,45 +381,97 @@
       "cacheVariables": {}
     },
     {
-      "name": "ninja-debug-valgrind-basic",
+      "name": "ninja-debug-emscripten",
       "inherits": [
         "base-debug",
-        "features-basic",
-        "features-valgrind"
+        "features-emscripten"
       ],
-      "displayName": "Debug build for Valgrind with reduced dependencies",
+      "displayName": "Debug build which builds an Emscripten library",
       "cacheVariables": {}
     },
     {
-      "name": "ninja-debug-valgrind",
+      "name": "ninja-debug-valgrind-minimal",
       "inherits": [
-        "base-debug",
-        "features-main",
-        "features-valgrind"
+        "features-valgrind",
+        "ninja-debug-minimal"
       ],
-      "displayName": "Debug build for Valgrind with more optional components",
+      "displayName": "Debug build for Valgrind without anything enabled",
       "cacheVariables": {}
     },
     {
-      "name": "ninja-debug-valgrind-minimal",
+      "name": "ninja-debug-valgrind-basic",
       "inherits": [
-        "base-debug",
-        "features-minimal",
-        "features-valgrind"
+        "features-valgrind",
+        "ninja-debug-basic"
       ],
-      "displayName": "Debug build for Valgrind without anything enabled",
+      "displayName": "Debug build for Valgrind with tests and reduced dependencies",
+      "cacheVariables": {}
+    },
+    {
+      "name": "ninja-debug-valgrind",
+      "inherits": [
+        "features-valgrind",
+        "ninja-debug"
+      ],
+      "displayName": "Debug build for Valgrind with tests and more optional components",
       "cacheVariables": {}
     },
     {
       "name": "ninja-debug-valgrind-maximal",
       "inherits": [
-        "base-debug",
-        "features-maximal",
-        "features-valgrind"
+        "features-valgrind",
+        "ninja-debug-maximal"
+      ],
+      "displayName": "Debug build for Valgrind with tests and everything enabled",
+      "cacheVariables": {}
+    },
+    {
+      "name": "ninja-debug-asan",
+      "inherits": [
+        "_allocator-none",
+        "ninja-debug",
+        "sanitizer-asan"
+      ],
+      "displayName": "Debug build for ASAN with tests and more optional components",
+      "cacheVariables": {}
+    },
+    {
+      "name": "ninja-debug-tsan",
+      "inherits": [
+        "_allocator-none",
+        "ninja-debug",
+        "sanitizer-tsan"
+      ],
+      "displayName": "Debug build for TSAN with tests and more optional components",
+      "cacheVariables": {}
+    },
+    {
+      "name": "ninja-debug-ubsan",
+      "inherits": [
+        "_allocator-none",
+        "ninja-debug",
+        "sanitizer-ubsan"
       ],
-      "displayName": "Debug build for Valgrind with everything enabled",
+      "displayName": "Debug build for UBSAN with tests and more optional components",
       "cacheVariables": {}
     },
+    {
```

Comment:
Moved from the bottom of the file to after `sanitizer` group because it also belongs to `debug` group.